File: Frontend/e-auction/src/Services/BidsService.js
import axios from "axios";

class bidServices{

    getAllBids(){
        return axios.get('http://localhost:9090/api/bids/');
    }

    // creaetBids(bid, sellerId){
    //     return axios.post('http://localhost:9090/api/seller/' + sellerId +'/bids', bid);
    // }

    deletebids(bidId){
        return axios.delete('http://localhost:9090/api/bids/{bidId}', bidId);
    }

    createBids(bid, buyerId, productId){
        console.log(productId);
        return axios.post('http://localhost:9090/api/buyer/' + buyerId +'/product/'+ productId + '/bids', bid);
    }
}

export default new bidServices();

File: Backend/E-Auction/src/main/java/com/subham/auction/service/BidsService.java
package com.subham.auction.service;

import java.util.List;

import com.subham.auction.payloads.BidsDto;

public interface BidsService {

	//create bids
	BidsDto createBids (BidsDto bidsDto, Long buyerId, Long productId);
	
	//update bids
	BidsDto updateBids (BidsDto bidsDto, Long bidId);
	
	//delete bid
	void deleteBid(Long bidId);
	
	//get all bids
	List<BidsDto> getAllBids();
	
	//get all bids by buyers
	List<BidsDto> getBidsByBuyer(Long buyerId);
	
	//get all bids by product
	List<BidsDto> getBidsByProduct(Long productId);
}
